src/client/components/configuration/categorySelector.ts

```diff
@@ -1,13 +1,199 @@
+
+    if (newName.includes(' ') || newName.includes(',')) {
+      this.#hasError = true
+      this.requestUpdate()
+      return
+    }
+
+    const event = new CustomEvent<IUpdatedCategory>('onCategoryUpdated', {
+      detail: { before: this.category, after: newName },
+    })
+    this.dispatchEvent(event)
+    this.#isEditing = false
+    this.requestUpdate()
+  }
+
+  #renderEditMode() {
+    return html`<div class="item-container editMode">
+      <vscode-text-field
+        style="${this.#hasError ? 'border: 1px solid red' : ''}"
+        value="${this.category}"
+        placeholder=${this.category}
+        @change=${this.#onCategoryNameChange}
+      >
+      </vscode-text-field>
+      <ul class="actions-container editMode" role="toolbar">
+        <li
+          class="action-item"
+          style="padding: 0 2px 0 0"
+          role="presentation"
+          title="Edit Category Item"
+        >
+          <vscode-button appearance="primary" @click="${this.#onCategoryNameChange}">
+            <label>OK</label>
+          </vscode-button>
+        </li>
+        <li
+          class="action-item"
+          style="padding: 0 0 0px 2px"
+          role="presentation"
+          title="Remove Category Item"
+        >
+          <vscode-button
+            appearance="secondary"
+            @click="${() => {
+              this.#isEditing = false
+              this.requestUpdate()
+            }}"
+          >
+            <label>Cancel</label>
+          </vscode-button>
+        </li>
+      </ul>
+    </div>`
+  }
+
+  #renderViewMode() {
+    return html`<div class="item-container">
+      <pre>${this.category}</pre>
+      <ul class="actions-container" role="toolbar">
+        <li class="action-item" role="presentation" title="Edit Category Item">
+          <a
+            class="action-label codicon codicon-settings-edit"
+            role="button"
+            aria-label="Edit Category Item"
+            aria-checked=""
+            tabindex="0"
+            @click="${() => {
+              this.#isEditing = true
+              this.requestUpdate()
+            }}"
+          >
+            <svg
+              width="16"
+              height="16"
+              viewBox="0 0 16 16"
+              xmlns="http://www.w3.org/2000/svg"
+              fill="currentColor"
+            >
+              <path
+                d="M13.23 1h-1.46L3.52 9.25l-.16.22L1 13.59 2.41 15l4.12-2.36.22-.16L15 4.23V2.77L13.23 1zM2.41 13.59l1.51-3 1.45 1.45-2.96 1.55zm3.83-2.06L4.47 9.76l8-8 1.77 1.77-8 8z"
+              />
+            </svg>
+          </a>
+        </li>
+        <li class="action-item" role="presentation" title="Remove Category Item">
+          <a
+            class="action-label codicon codicon-settings-remove"
+            role="button"
+            aria-label="Remove Category Item"
+            aria-checked=""
+            @click="${() => {
+              const event = new CustomEvent('onCategoryRemoved', {
```

Comment:
can we initialize the Custom Events at the constructor level ? and then do something like:
`this.dispatchEvent(this.onCategoryRemovedEvent)`